### PR TITLE
Expose doc for SLO Corrections 

### PR DESCRIPTION
**Pull request by zhengshizhao**
SLO Status Corrections are being released to public beta, we want to expose docs for SLO correction resource. 
reference: [MA-2184](https://datadoghq.atlassian.net/browse/MA-2184)

### CODE REVIEW

**Comment by zhengshizhao** [comment]
> 👋 Could you update and remove the following lines as well. We currently remove the generated slo corrections doc file:
> 
> https://github.com/DataDog/terraform-provider-datadog/blob/b892d6c2a5a13ec13e4c7063c09f55b6afa066b1/scripts/generate-docs.sh#L24-L25

removed~

**Review by iminoso — DISMISSED** [review]

**Review by iminoso — APPROVED** [review]

File: docs/resources/slo_correction.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+---
+
+# datadog_slo_correction (Resource)
+
+Resource for interacting with the slo_correction API
+
+## Example Usage
+
+```terraform
+# Create a new Datadog SLO Correction. slo_id can be derived from slo resource or specify an slo id of an existing SLO.
```
Comment (NouemanKHAL) [inline docs/resources/slo_correction.md:16]:
```suggestion
# Create a new Datadog SLO correction. slo_id can be derived from slo resource or specify an slo id of an existing SLO.
```
I think it's supposed to be lowercase, right?

File: docs/resources/slo_correction.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    timeframe = "7d"
+    target    = 99.5
+    warning   = 99.8
+  }
+  tags = ["foo:bar"]
+}
+resource "datadog_slo_correction" "example_slo_correction" {
+  category    = "Scheduled Maintenance"
+  description = "correction example"
+  end         = 1735718600
```
Comment (athap) [inline docs/resources/slo_correction.md:37]: I wonder if  for documentation there is a way to add a comment next to timestamp (like human readable value)

File: docs/resources/slo_correction.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  slo_id      = "datadog_service_level_objective.example_slo.id"
+  start       = 1735707000
```
Comment (athap) [inline docs/resources/slo_correction.md:39]: nit - can we move `start` before `end`?

File: docs/resources/slo_correction.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- **category** (String) Category the SLO correction belongs to
+- **end** (Number) Ending time of the correction in epoch seconds
```
Comment (athap) [inline docs/resources/slo_correction.md:50]: maybe `Ending time` --> `End time`

File: docs/resources/slo_correction.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+- **slo_id** (String) ID of the SLO that this correction will be applied to
+- **start** (Number) Starting time of the correction in epoch seconds
```
Comment (athap) [inline docs/resources/slo_correction.md:52]: maybe `Starting time` --> `Start time`

**Review by athap — APPROVED** [review]
couple of nits but lgtm

File: docs/resources/slo_correction.md (hunk 5 above)
Comment (zhengshizhao) [reply docs/resources/slo_correction.md:52]: same as end

File: docs/resources/slo_correction.md (hunk 4 above)
Comment (zhengshizhao) [reply docs/resources/slo_correction.md:50]: This is consistent with other API clients /doc page: https://docs.datadoghq.com/api/latest/service-level-objective-corrections/

File: docs/resources/slo_correction.md (hunk 2 above)
Comment (zhengshizhao) [reply docs/resources/slo_correction.md:37]: do we need comment here in the terraform example as we already have description for start/end in epoch seconds?

**Review by therve — APPROVED** [review]

**Pull request merged**